src/StackExchange.Redis.Extensions.Core/StackExchangeRedisCacheClient.cs

```diff
@@ -628,8 +628,13 @@ public IEnumerable<string> SearchKeys(string pattern)
 
 			foreach (var endpoint in endPoints)
 			{
-				var dbKeys = Database.Multiplexer.GetServer(endpoint).Keys(Database.Database, pattern);
+				var server = Database.Multiplexer.GetServer(endpoint);
+				if (!server.IsConnected || !server.Features.Scan)

```

Comment:
Hi all:
Any traction on this?
Maybe we can split the statement into two like:
`var isServerEligible = true;
if (!server.IsConnected) {
    //log here error/critical
    isServerEligible = false;
}
if (!server.Features.Scan) {
    //log here only only when verbose mode or not at all
    isServerEligible = false;
}
if (!isServerEligible) {
    continue;
}`

Will this better accomplish the intention?